crates/stages/src/stages/bodies.rs

```diff
@@ -238,15 +215,16 @@ impl<D: BodyDownloader, C: Consensus> BodyStage<D, C> {
 
         let mut bodies_to_download = Vec::new();
         while let Some(Ok((block_number, header_hash))) = walker.next() {
-            let header = header_cursor
-                .seek_exact((block_number, header_hash).into())?
-                .ok_or(DatabaseIntegrityError::Header { number: block_number, hash: header_hash })?
-                .1;
+            let (_, header) = header_cursor.seek_exact((block_number, header_hash).into())?.ok_or(
+                DatabaseIntegrityError::Header { number: block_number, hash: header_hash },
+            )?;
+
             if header.ommers_hash == EMPTY_LIST_HASH && header.transactions_root == EMPTY_ROOT {
+                // TODO: fix this
```

Comment:
If we indeed move to the new changeset structure let's not forget to add a note that the gaps issue with the returned empty bodies stream is no longer present

Reply:
@rkrasiuk pls add the above comment in the code to ensure we don't lose the context around this todo